From 04ec1d24692f66da1fbaeae276b28ec74e851eec Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Sun, 2 Oct 2022 12:36:13 +0200
Subject: [PATCH 42/52] Revert "elevator: remove redundant code in
 elv_unregister_queue()"

This reverts commit 9b431fae8f8a89566d63adca1c89ea8c92af364c.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/elevator.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/block/elevator.c b/block/elevator.c
index 0a72d6fbb..7cb61820c 100644
--- a/block/elevator.c
+++ b/block/elevator.c
@@ -524,6 +524,8 @@ void elv_unregister_queue(struct request_queue *q)
 	lockdep_assert_held(&q->sysfs_lock);
 
 	if (e && test_and_clear_bit(ELEVATOR_FLAG_REGISTERED, &e->flags)) {
+		struct elevator_queue *e = q->elevator;
+
 		kobject_uevent(&e->kobj, KOBJ_REMOVE);
 		kobject_del(&e->kobj);
 	}
-- 
2.38.0.rc1.6.g4fd6c5e444

